staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator.go

```diff
@@ -53,6 +53,12 @@ type specAggregator struct {
 	// provided for dynamic OpenAPI spec
 	openAPIService          *handler.OpenAPIService
 	openAPIVersionedService *handler.OpenAPIService
+
+	// Record delegation API services' name. If delegation API services' OpenAPI
+	// spec is dynamic (e.g. apiExtensionsServer), the recorded delegationAPIServices
+	// can be exposed to aggregation controller, which periodically try syncing the
+	// OpenAPI specs based on the API services' name as key
+	delegationAPIServices []string
```

Comment:
We have two names for the same thing: local and delegation api service, don't we?

Reply:
Right. "Local" APIServices contain delegation APIServices plus an extra [static spec with no handler](https://github.com/kubernetes/kubernetes/blob/f48e18faa4dc035cc927c6a2b34c83c8475b55fa/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator.go#L87-L88) IMO. 

I think we should be able to drop the `delegationAPIServices` slice as you pointed out in the other comment https://github.com/kubernetes/kubernetes/pull/67205#discussion_r233735114